test/models/parser/twitter_profile_test.rb

```diff
@@ -71,62 +60,123 @@ def empty_doc
     assert_equal true, match_four.is_a?(Parser::TwitterProfile)
   end
 
-  test "should send Twitter error to Errbit and return default values" do
-    Twitter::REST::Client.any_instance.stubs(:user).raises(Twitter::Error)
+  test "it makes a get request to the user lookup by username endpoint successfully" do
+    stub_configs({'twitter_bearer_token' => 'test' })
+    
+    WebMock.stub_request(:get, "https://api.twitter.com/2/users/by")
+      .with(query: query)
+      .with(headers: { "Authorization": "Bearer test" })
+      .to_return(status: 200, body: response_fixture_from_file('twitter-profile-response-success.json'))
+
+    data = Parser::TwitterProfile.new('https://m.twitter.com/fake_user').parse_data(empty_doc)
+    
+    assert_equal '2009-04-07T15:40:56.000Z', data['published_at']
+    assert_equal '@fake_user', data['username']
+  end
+
+  test "it makes a get request to the user lookup by username endpoint and raises an error when 401 is returned" do
+    stub_configs({'twitter_bearer_token' => 'test' })
+
+    WebMock.stub_request(:get, "https://api.twitter.com/2/users/by")
+      .with(query: query)
+      .with(headers: { "Authorization": "Bearer test" })
+      .to_return(status: 401)
+
+    assert_raises ProviderTwitter::ApiError do
+      Parser::TwitterProfile.new('https://m.twitter.com/fake_user').parse_data(empty_doc)
+    end
+  end
+
+  test "it makes a get request to the user lookup by username endpoint, raises an error when 500 is returned and notifies sentry" do
+    stub_configs({'twitter_bearer_token' => 'test' })
+
+    WebMock.stub_request(:get, "https://api.twitter.com/2/users/by")
+      .with(query: query)
+      .with(headers: { "Authorization": "Bearer test" })
+      .to_return(status: 500, body: response_fixture_from_file('twitter-profile-response-error.json'))
 
-    data = {}
     sentry_call_count = 0
     arguments_checker = Proc.new do |e|
       sentry_call_count += 1
-      assert_equal Twitter::Error, e.class
     end
-
+    
     PenderSentry.stub(:notify, arguments_checker) do
-      data = Parser::TwitterProfile.new('https://www.twitter.com/fakeaccount').parse_data(nil)
+      assert_raises ProviderTwitter::ApiError do
```

Comment:
same comment here about not wanting to return an error from parse_data, though we do want to check that we send to Sentry